cola/linalg/decompositions/decompositions.py

```diff
@@ -78,6 +98,31 @@ def __call__(self, A: LinearOperator):
         return arnoldi(A, **self.__dict__)
 
 
+@dataclass
+class LanczosSVD(Algorithm):
+    """
+    Does the SVD using the Lanczos decomposition,
+    Args:
+        start_vector (Array, optional): (n,) or (n, b) vector to start the algorithm.
+        max_iters (int, optional): The maximum number of iterations to run.
+        tol (float, optional): Relative error tolerance.
+        pbar (bool, optional): Whether to show progress bar.
+        key (PRNGKey, optional): Random key to use for the algorithm.
+            PRNGKey for jax, long integer for numpy or pytorch.
+    """
+    start_vector: Array = None
+    max_iters: int = 1_000
+    tol: float = 1e-6
+    pbar: bool = False
+    key: Optional[PRNGKey] = None
+
+    def __call__(self, A: LinearOperator):
+        Sigma0_2, V, _ = lanczos_eigs(A.H @ A, **self.__dict__)
```

Comment:
I think you can do it with one call, get V and then A@V should give the other singular vectors (when normalized)

Reply:
Good point.